src/rpc/core_rpc_server.cpp

```diff
@@ -61,8 +64,43 @@ using namespace epee;
 
 #define OUTPUT_HISTOGRAM_RECENT_CUTOFF_RESTRICTION (3 * 86400) // 3 days max, the wallet requests 1.8 days
 
+#define DEFAULT_PAYMENT_DIFFICULTY 1000
+#define DEFAULT_PAYMENT_CREDITS_PER_HASH 100
+
+#define RPC_TRACKER(rpc) \
+  PERF_TIMER(rpc); \
+  RPCTracker tracker(#rpc)
+
 namespace
 {
+  class RPCTracker
+  {
+  public:
+    RPCTracker(const char *rpc): rpc(rpc), timer(rpc, MONERO_DEFAULT_LOG_CATEGORY, 1000000, tools::performance_timer_log_level) {
+    }
+    ~RPCTracker() {
+      boost::unique_lock<boost::mutex> lock(mutex);
+      auto &e = tracker[rpc];
+      std::get<0>(e)++;
+      std::get<1>(e) = timer.value();
+    }
+    void pay(uint64_t amount) {
+      boost::unique_lock<boost::mutex> lock(mutex);
+      auto &e = tracker[rpc];
+      std::get<2>(e) += amount;
+    }
+    const std::string &rpc_name() const { return rpc; }
+    static void clear() { boost::unique_lock<boost::mutex> lock(mutex); tracker.clear(); }
+    static std::unordered_map<std::string, std::tuple<uint64_t, uint64_t, uint64_t>> data() { boost::unique_lock<boost::mutex> lock(mutex); return tracker; }
```

Comment:
A tuple is kind of confusing this context, because there is no indication what each `uint64_t` is representing.

Reply:
Fair, I'll make it a struct. It... grew.